doc/semantics-of-the-publishing-api.md

```diff
@@ -0,0 +1,406 @@
+## Publishing API V2 Semantics
+
+This document explains the semantics of Publishing API V2. This includes:
+
+- What do each of the fields mean
+- What values are allowed in each field
+- What is the effect of choosing one value over another
+
+There is another document that explains the syntax for how the API should be
+used. That document can be found
+[here](https://docs.google.com/document/d/1zYFZFZ2TLzasgX4SZTxuQhLzqRYCk4hWKBTcb8QFIRA/edit).
```

Comment:
The syntax docs are appended to @elliotcm's overview [on the atlassian wiki](https://gov-uk.atlassian.net/wiki/display/TECH/Publishing+Platform). IMO these would be better placed in GH along with examples and semantics, any objections?